### PR TITLE
New S2 Layer Catalog (Experimental)

### PR DESCRIPTION
**Pull request by ibgreen**
# For #2588

### CODE REVIEW

File: modules/s2-layers/src/s2-layer/s2-layer.js [hunk 1]
```diff
@@ -0,0 +1,138 @@
...
+};
+
+export default class S2Layer extends CompositeLayer {
+  renderLayers() {
+    // Layer prop
+    const {data, getS2Token} = this.props;
+
+    // Rendering props underlying layer
+    const {
+      elevationScale,
```
Comment (ibgreen) [inline modules/s2-layers/src/s2-layer/s2-layer.js]: Do we need to list all these props, can't we just forward the props object with some overrides?

What is our standard approach here?

File: docs/layers/s2-layers/s2-layer.md [hunk 2]
```diff
@@ -0,0 +1,89 @@
+<p class="badges">
+  <img src="https://img.shields.io/badge/64--bit-support-blue.svg?style=flat-square" alt="64-bit" />
+  <img src="https://img.shields.io/badge/extruded-yes-blue.svg?style=flat-square" alt="extruded" />
+</p>
+
+# S2Layer
```
Comment (ibgreen) [inline docs/layers/s2-layers/s2-layer.md]: Add `(Experimental)`

File: modules/s2-layers/src/s2-layer/s2-layer.js (hunk 1 above)
Comment (xintongxia) [reply modules/s2-layers/src/s2-layer/s2-layer.js]: yes. Other layers also did the same, like `PolygonLayer`.

File: docs/layers/s2-layers/s2-layer.md [hunk 3]
```diff
@@ -0,0 +1,93 @@
+<p class="badges">
+  <img src="https://img.shields.io/badge/64--bit-support-blue.svg?style=flat-square" alt="64-bit" />
+  <img src="https://img.shields.io/badge/extruded-yes-blue.svg?style=flat-square" alt="extruded" />
+</p>
+
+# S2Layer (Experimental)
+
+The S2Layer renders filled and/or stroked polygons, with geometry automatically calculated based on an S2 token (geospatial index).
```
Comment (ibgreen) [inline docs/layers/s2-layers/s2-layer.md]: Recommend adding a link to http://s2geometry.io/ the first time it is mentioned on a page.

File: docs/layers/s2-layers/s2-layer.md [hunk 4]
```diff
@@ -0,0 +1,93 @@
...
+
+Accepts all render options from `PolygonLayer`.
+
+### Data Accessors
+
+Accepts all accessors from `PolygonLayer`, except `getPolygon` has been replaced with `getToken`:
+
+##### `getToken` (Function, optional) ![transition-enabled](https://img.shields.io/badge/transition-enabled-green.svg?style=flat-square")
+
+* default: `object => object.token`
```
Comment (ibgreen) [inline docs/layers/s2-layers/s2-layer.md]: A bit more info here, a link to the S2 site page?

File: docs/layers/s2-layers/s2-layer.md [hunk 5]
```diff
@@ -0,0 +1,93 @@
...
+const App = ({data, viewport}) => {
+
+  /**
+   * Data format:
+   * [
+   *   {
+   *     // S2 Cell in SF Bay Area
+   *     token: "80858004",
+   *     // Simple polygon (array of coords)
+   *     token: [[-122.4, 37.7], [-122.4, 37.8], [-122.5, 37.8], [-122.5, 37.7], [-122.4, 37.7]],
```
Comment (ibgreen) [inline docs/layers/s2-layers/s2-layer.md]: I think this comment is a little confusing. The token is passed to the S2 library which generates that polygon. Can we rephrase a little, or cut the polygon?

File: docs/layers/s2-layers/s2-layer.md [hunk 6]
```diff
@@ -0,0 +1,93 @@
...
+
+Accepts all accessors from `PolygonLayer`, except `getPolygon` has been replaced with `getToken`:
+
+##### `getToken` (Function, optional) ![transition-enabled](https://img.shields.io/badge/transition-enabled-green.svg?style=flat-square")
+
+* default: `object => object.token`
+
+## Remarks
+
+* Uses the [`s2-geometry`](http://s2geometry.io/) library for S2 polygon calculations.
```
Comment (ibgreen) [inline docs/layers/s2-layers/s2-layer.md]: I think this remark should be moved to the top

**Pull request merged**